bot/exts/evergreen/status_codes.py

```diff
@@ -20,62 +26,53 @@ class HTTPStatusCodes(commands.Cog):
-                else:
-                    raise NotImplementedError
-
-        except ValueError:
-            embed.set_footer(text="Inputted status code does not exist.")
-
-        except NotImplementedError:
-            embed.set_footer(text="Inputted status code is not implemented by http.cat yet.")
-
-        finally:
-            await ctx.send(embed=embed)
+        """Send a cat version of the requested HTTP status code."""
+        if code in range(100, 600):
+            await self.build_embed(url=HTTP_CAT_URL.format(code=code), ctx=ctx, code=code)
+            return
+        await ctx.send(embed=ERROR_LENGTH_EMBED)
 
     @http_status_group.command(name="dog")
     async def http_dog(self, ctx: commands.Context, code: int) -> None:
-        """Sends an embed with an image of a dog, portraying the status code."""
-        # These codes aren't server-friendly.
-        if code in (304, 422):
-            await self.http_cat(ctx, code)
+        """Send a dog version of the requested HTTP status code."""
+        if code in range(100, 600):
+            await self.build_embed(url=HTTP_DOG_URL.format(code=code), ctx=ctx, code=code)
             return
-
-        embed = discord.Embed(title=f"**Status: {code}**")
-        url = HTTP_DOG_URL.format(code=code)
-
-        try:
-            HTTPStatus(code)
-            async with self.bot.http_session.get(url, allow_redirects=False) as response:
-                if response.status != 302:
-                    embed.set_image(url=url)
-                else:
-                    raise NotImplementedError
-
-        except ValueError:
-            embed.set_footer(text="Inputted status code does not exist.")
-
-        except NotImplementedError:
-            embed.set_footer(text="Inputted status code is not implemented by httpstatusdogs.com yet.")
-
-        finally:
-            await ctx.send(embed=embed)
+        await ctx.send(embed=ERROR_LENGTH_EMBED)
+
+    async def build_embed(self, url: str, ctx: commands.Context, code: int) -> None:
+        """Attempt to build and dispatch embed. Append error message instead if something goes wrong."""
+        async with self.bot.http_session.get(url, allow_redirects=False) as response:
+            if response.status in range(200, 300):
+                await ctx.send(
+                    embed=discord.Embed(
+                        title=STATUS_TEMPLATE.format(code=code)
+                    ).set_image(url=url)
+                )
+            elif response.status == 404:
+                await ctx.send(
+                    embed=discord.Embed(
+                        title=ERR_404.format(code=code)
+                    ).set_image(url=url)
+                )
+            else:
+                await ctx.send(
+                    embed=discord.Embed(
+                        title=STATUS_TEMPLATE.format(code=code)
+                    ).Embed.set_footer(ERR_UNKNOWN.format(code=code))
+                )
```

Comment:
This doesn't seem quite right at the moment, the `.Embed` shouldn't be that, and `ERR_UNKNOWN.format(code=code)` needs to be passed in as a keyword argument, so `set_footer(text=ERR_UNKNOWN.format(code=code))`.

What I noticed to get this result is that instead of returning 404 when the image isn't foundlike the cat url, httpstatusdogs.com seems to return a 302, meaning it isn't caught in the `elif` and gets to the else. We could probably change line 64 to check for 302 first, `elif response.status in (302, 404):`, maybe with a comment as to why this is necessary as well.

This is the result I got after fixing the `else`, catching the 302 in the `elif` should make the responses more consistent:

![Discord_TDrt14y8HD](https://user-images.githubusercontent.com/22353562/131559462-00bbec69-44d3-4289-a3a7-364523914ba9.png)